internal/rundata/dut.go

```diff
@@ -0,0 +1,251 @@
+	const notSupported = `
+	if di.model != "" {
+		modelPath := ocpath.Root().System().Model()
+		model, err := ygnmi.Get(ctx, y.Client, modelPath.State())
+		if err != nil {
+			glog.Errorf("Missing system model: %v", err)
+		} else {
+			di.model = model
+			glog.V(2).Infof("Setting model from system model: %s", di.model)
+		}
+	}
+
+	if di.osver != "" {
+		softVerPath := ocpath.Root().System().SoftwareVersion()
+		softVer, err := ygnmi.Get(ctx, y.Client, softVerPath.State())
+		if err != nil {
+			glog.Errorf("Missing system software-version: %v", err)
+		} else {
+			di.osver = softVer
+			glog.V(2).Infof("Setting osver from system software-version: %s", di.osver)
+		}
+	}
+`
+}
+
+// shortVendor canonicalizes full vendor string in short uppercase form, if possible.
+func (di *dutInfo) shortVendor() string {
+	if di.vendor == "" {
+		return ""
+	}
+	vendors := []ondatra.Vendor{
+		ondatra.ARISTA, ondatra.CISCO, ondatra.DELL, ondatra.JUNIPER, ondatra.IXIA, ondatra.CIENA,
+		ondatra.PALOALTO, ondatra.ZPE, ondatra.NOKIA,
+	}
+	fullUpper := strings.ToUpper(di.vendor)
+	for _, vendor := range vendors {
+		shortUpper := strings.ToUpper(vendor.String())
+		if strings.Contains(fullUpper, shortUpper) {
+			return shortUpper
+		}
+	}
+	if strings.Contains(fullUpper, "PALO ALTO") {
+		return "PALOALTO"
+	}
+	return strings.SplitN(fullUpper, " ", 2)[0]
+}
+
+// ciscoRE reduces model string from e.g. "Cisco xxxx n-slot Chassis" to just "xxxx".
+var ciscoRE = regexp.MustCompile(`Cisco (.*?) .*`)
+
+// jnpRE reduces model string from e.g. "JNP10008 [PTX10008]" to just "PTX10008".
+var jnpRE = regexp.MustCompile(`JNP.* \[(.*)\]`)
+
+// shortModel canonicalizes full model to short form.
+func (di *dutInfo) shortModel() string {
+	if matches := ciscoRE.FindStringSubmatch(di.model); len(matches) >= 2 {
+		return matches[1]
+	}
+	if matches := jnpRE.FindStringSubmatch(di.model); len(matches) >= 2 {
+		return matches[1]
+	}
+	return di.model
+}
+
+// put exports the dutInfo to a map with the given dut ID.
+func (di *dutInfo) put(m map[string]string, id string) {
+	if di.vendor != "" {
+		m[id+".vendor.full"] = di.vendor
+		m[id+".vendor"] = di.shortVendor()
+	}
+	if di.model != "" {
+		m[id+".model.full"] = di.model
+		m[id+".model"] = di.shortModel()
+	}
+	if di.osver != "" {
+		m[id+".os_version"] = di.osver
+	}
+}
+
+// newDUTInfo creates a newly populated dutInfo.
+func newDUTInfo(ctx context.Context, y components.Y) *dutInfo {
+	di := &dutInfo{}
+	di.setFromSystem(ctx, y)
+	di.setFromComponentChassis(ctx, y)
+	di.setFromComponentOS(ctx, y)
+	di.setFromLLDP(ctx, y)
+	return di
+}
+
+// dutsInfo populates the DUT properties for all DUTs in the reservation.
+func dutsInfo(ctx context.Context, m map[string]string, resv *binding.Reservation) {
+	for id, dut := range resv.DUTs {
+		gnmic, err := dut.DialGNMI(ctx)
+		if err != nil {
+			glog.Errorf("Could not dial GNMI to dut %s: %v", dut.Name(), err)
+			continue
+		}
+		yc, err := ygnmi.NewClient(gnmic)
```

Comment:
Probably better to add two functions in components`New()` and `NewWithT()` where `New()` returns `(Y, error)`?

Reply:
The T version doesn't need any struct wrapper since they can just use *ondatra.DUTDevice. Also, the T version is still used by existing tests, so I don't want to have to change them in this PR.